src/Algolia.Search/Models/Settings/IndexSettings.cs

```diff
@@ -281,20 +281,44 @@ private List<string> Slaves
         public object RemoveStopWords { get; set; }
 
         // performance
+        [JsonIgnore]
+        private bool _numericAttributesForFilteringWasSet = false;
+
+        /// <summary>
+        /// ShouldSerialize predicate from NewtonSoft.json to avoid serializing empty field.
+        /// https://www.newtonsoft.com/json/help/html/ConditionalProperties.htm
+        /// </summary>
+        public bool ShouldSerializeNumericAttributesForFiltering()
+        {
+            return _numericAttributesForFilteringWasSet;
+        }
+
+        private List<string> _numericAttributesForFiltering;
```

Comment:
question: shouldn't we add `[JsonIgnore]` here? Can you make sure that `_numericAttributesForFiltering` isn't send in the payload? If yes we're good to merge.

Reply:
I've just check and this private field isn't sent in the payload 🥳